### PR TITLE
Update user guide 

### PR DESCRIPTION
**Pull request by techjay-c**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -121,94 +121,108 @@ who has 6 years of experience and specializing in 'braces' into the list of dent
...
 
 <div markdown="span" class="alert alert-primary">:bulb: **Tip:**
-To check the Dentist ID for a dentist, you can simply enter the command `list-dentist`.
+To check the Dentist ID of a dentist, you can simply enter the command `list-dentist`.
 </div>
 
+Examples:
+* `delete-dentist 6` deletes the dentist with the `DENTIST_ID` 6.
+
+**Format:** `delete-dentist DENTIST_ID`
```
Comment (ruth-lim) [inline docs/UserGuide.md:152]: Format is repeated twice in this section

File: docs/UserGuide.md [hunk 2]
```diff
@@ -121,94 +121,108 @@ who has 6 years of experience and specializing in 'braces' into the list of dent
...
+
+### Searches for a dentist by Dentist ID or keyword: `search-dentist`
```
Comment (ruth-lim) [inline docs/UserGuide.md:154]: Standardise the verb form with other commands? Are we going with -ing/ -s?

File: docs/UserGuide.md [hunk 3]
```diff
@@ -121,94 +121,108 @@ who has 6 years of experience and specializing in 'braces' into the list of dent
...
+* `search-dentist 2` searches for the dentist with the `DENTIST_ID` 2.
 
-### Searching a dentist by name: `search-dentist NAME`
+### Filter dentist: `filter-dentist`
 
-Searches for dentists whose name contains `NAME`  in ToothTracker. 
-**This search may return more than one dentist.** This is helpful when:
-* You want quick access to one or more specific Dentists particulars, **but you forgot the `Dentist ID` of the dentists** you are searching for.
+Narrows down your search for dentists based on a specified attribute. When this command is used, there might be more than
+one result that matches your search criteria.** This is helpful when:
```
Comment (ruth-lim) [inline docs/UserGuide.md:181]: Is there extra ** here?

File: docs/UserGuide.md [hunk 4]
```diff
@@ -121,94 +121,108 @@ who has 6 years of experience and specializing in 'braces' into the list of dent
...
+To check the Dentist ID for a dentist, you can simply enter the command `list-dentist`.
+</div>
 
+<div markdown="block" class="alert alert-info">
+**:information_source: Notes about editing tags:**<br>
+When editing tags, you have to include any previous tags that was already included in the dentist, or else these tags will be removed. 
+This also means that you can use `t/` to remove all tags from a dentist using the <code>edit-dentist</code> command.
+</div>
 
 ## Part 2 - Patient Features
```
Comment (ruth-lim) [inline docs/UserGuide.md:227]: 1. Under Examples: command should be in lower caps `add-patient`
2. Can double check the order of the parameters for the format of `add-patient`? I think we should standardise with the order we give in the example.

File: docs/UserGuide.md [hunk 5]
```diff
@@ -282,14 +294,30 @@ You may use `list-patient` to check out the patient's ID first.
...
 
 Examples:
 * `search-patient Thomas` searches for patients with the name containing the keyword `Thomas`.
 * `search-patient 88` searches for a patient with the ID `88`. If one or more matching patients are found based on
   your search criteria, the system will list the matching patients along with their details.
 
+### Filter patient: `filter-patient`
+
+Narrows down your search for patients based on a specified attribute. When this command is used, there might be more than
+one result that matches your search criteria.** This is helpful when:
```
Comment (ruth-lim) [inline docs/UserGuide.md:308]: Extra ** here?

File: docs/UserGuide.md [hunk 6]
```diff
@@ -457,29 +485,31 @@ Afterwards, you may edit the particulars using `edit-dentist` or `edit-patient`
...
+| **Delete Dentist**                      | `delete-dentist DENTIST_ID`<br> e.g., `delete-dentist 3`                                                                                                                                                                                       |
+| **List Dentist**                        | `list-dentist`                                                                                                                                                                                                                                 |
+| **Search Dentist by Dentist ID**        | `search-dentist DENTIST_ID` <br> e.g., `search-dentist 2`                                                                                                                                                                                      |
+| **Search Dentist by Name**              | `search-dentist NAME` <br> e.g., `search-dentist Tom`                                                                                                                                                                                          |
```
Comment (ruth-lim) [inline docs/UserGuide.md:495]: Change to Search Dentist by Keyword?

File: docs/UserGuide.md [hunk 7]
```diff
@@ -457,29 +485,31 @@ Afterwards, you may edit the particulars using `edit-dentist` or `edit-patient`
...
+| **Delete Patient**                      | `delete-patient PATIENT_ID`<br> e.g., `delete-patient 3`                                                                                                                                                                                       |
+| **List Patients**                       | `list-patient`                                                                                                                                                                                                                                 |
+| **Search Patient by Patient ID**        | `search-patient PATIENT_ID`  <br> e.g., `search-patient 3`                                                                                                                                                                                     |
+| **Search Patient by Name**              | `search-patient NAME` <br> e.g., `search-patient John`                                                                                                                                                                                         |
```
Comment (ruth-lim) [inline docs/UserGuide.md:501]: Change to Search Patient by Keyword?

**Review by ruth-lim — CHANGES_REQUESTED** [review]

**Review by SynapseProgramming — APPROVED** [review]
approved.

**Review by ruth-lim — APPROVED** [review]
LGTM

**Pull request merged**